### PR TITLE
[WIP] Rewrite docs to include examples

### PR DESCRIPTION
**Pull request by timneutkens**
I rewrote / moved a lot of stuff around. Please have a look :) Will add the examples soon.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,83 +15,94 @@ _**Micro —** Async ES6 HTTP microservices_
 * **Simple**. Oriented for single purpose modules (function).
 * **Explicit**. No middleware. Modules declare all dependencies.
 * **Standard**. Just HTTP!
-* **Lightweight**. The package is small and the `async` transpilation fast and transparent
+* **Lightweight**. The package is small and the `async` transpilation is fast and transparent
 
-## Example
+## How to use
```
Comment (leo) [inline README.md]: > Usage

File: README.md [hunk 2]
```diff
@@ -15,83 +15,94 @@ _**Micro —** Async ES6 HTTP microservices_
...
-```js
-const {send} = require('micro')
-const sleep = require('then-sleep')
-
-module.exports = async function (req, res) {
-  await sleep(500)
-  send(res, 200, 'Ready!')
-}
+```
+npm install micro --save
```
Comment (leo) [inline README.md]: Command line indicator (dollar sign) missing. Also please put the flag before the package name!

File: README.md [hunk 3]
```diff
@@ -15,83 +15,94 @@ _**Micro —** Async ES6 HTTP microservices_
...
 ```
 
-To run the microservice on port `3000`, use the `micro` command:
+Add a script to your package.json like this:
```
Comment (leo) [inline README.md]: `package.json`, not package.json

**Review by leo — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (timneutkens) [reply README.md]: Done 🔥

File: README.md (hunk 2 above)
Comment (timneutkens) [reply README.md]: Done 🔥

File: README.md (hunk 1 above)
Comment (timneutkens) [reply README.md]: Done 🔥

**Comment by NetOpWibby** [comment]
Oh wow, thanks for this PR! I learned a lot by looking at the revised README.

File: README.md [hunk 4]
```diff
@@ -15,82 +15,96 @@ _**Micro —** Async ES6 HTTP microservices_
...
 
 ```bash
 micro sleep.js
 ```
-
-To run the microservice on port `3000` and localhost instead of listening on every interface, use the `micro` command:
-
-```bash
-micro -H localhost sleep.js
+{
```
Comment (onbjerg) [inline README.md]: Missing opening code fence (```)

File: README.md [hunk 5]
```diff
@@ -15,82 +15,96 @@ _**Micro —** Async ES6 HTTP microservices_
...
+const { json } = require('micro')
+
+module.exports = async function (req, res) {
+  const data = await json(req)
+  console.log(data.price)
+  return ''
+}
+```
+
+#### Api
```
Comment (onbjerg) [inline README.md]: API (it's an abbreviation)

File: README.md [hunk 6]
```diff
@@ -99,18 +113,23 @@ npm start
...
+```js
+const { send } = require('micro')
+module.exports = async function (req, res) {
+  const statusCode = 400
+  const data = { error: 'Custom error message' }
+  send(res, statusCode, data)
+}
+```
+
+#### Api
```
Comment (onbjerg) [inline README.md]: API (it's an abbreviation)

File: README.md [hunk 7]
```diff
@@ -122,62 +141,31 @@ npm start
...
+  await sleep(500)
+  return 'Hello world'
+})
 
-#### createError
+server.listen(3000)
+```
 
-**`createError(code, msg, orig)`**
+#### Api
```
Comment (onbjerg) [inline README.md]: API (it's an abbreviation)

File: README.md [hunk 8]
```diff
@@ -249,6 +238,24 @@ function handleErrors (fn) {
 }
 ```
 
+#### Api
```
Comment (onbjerg) [inline README.md]: API (it's an abbreviation)

**Comment by onbjerg** [comment]
Added some comments, hope you don't mind 😊

File: README.md (hunk 4 above)
Comment (timneutkens) [reply README.md]: Fixed 🙌

File: README.md (hunk 5 above)
Comment (timneutkens) [reply README.md]: Fixed 🙌

File: README.md (hunk 6 above)
Comment (timneutkens) [reply README.md]: Fixed 🙌

File: README.md (hunk 7 above)
Comment (timneutkens) [reply README.md]: Fixed 🙌

File: README.md (hunk 8 above)
Comment (timneutkens) [reply README.md]: Fixed 🙌

**Comment by timneutkens** [comment]
@leo @rauchg can you do a final review?

**Comment by timneutkens** [comment]
I wonder if we should disable `xo` for the examples 🤔

**Comment by leo** [comment]
@timneutkens 😍

**Pull request merged**